core/chain-configs/src/genesis_config.rs

```diff
@@ -468,37 +468,75 @@ impl GenesisJsonHasher {
     }
 }
 
+pub enum GenesisValidationMode {
+    Full,
+    UnsafeFast,
+}
+
 impl Genesis {
-    pub fn new(config: GenesisConfig, records: GenesisRecords) -> Self {
+    pub fn new(
+        config: GenesisConfig,
+        records: GenesisRecords,
+        genesis_validation: GenesisValidationMode,
+    ) -> Self {
         let genesis = Self { config, records, records_file: PathBuf::new() };
-        validate_genesis(&genesis);
+        match genesis_validation {
+            GenesisValidationMode::Full => {
+                validate_genesis(&genesis);
+            }
+            GenesisValidationMode::UnsafeFast => {
+                warn!("Skipped genesis validation");
+            }
+        }
         genesis
     }
 
-    pub fn new_with_path(config: GenesisConfig, records_file: PathBuf) -> Self {
+    pub fn new_with_path(
+        config: GenesisConfig,
+        records_file: PathBuf,
+        genesis_validation: GenesisValidationMode,
+    ) -> Self {
         let genesis = Self { config, records: GenesisRecords(vec![]), records_file };
-        validate_genesis(&genesis);
+        match genesis_validation {
+            GenesisValidationMode::Full => {
+                validate_genesis(&genesis);
+            }
+            GenesisValidationMode::UnsafeFast => {
+                warn!("Skipped genesis validation");
+            }
+        }
```

Comment:
Might make sense to dedupe the code. Usually it's best when there's one "primary constructor", everything else delegates to.